File: Server/src/main/java/core/game/content/dialogue/MonkeyDialogue.java
package core.game.content.dialogue;

import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * The dialogue for the desert monkey.
 * @author 'Vexia
 */
@Initializable
public final class MonkeyDialogue extends DialoguePlugin {

	/**
	 * Constructs a new {@code MonkeyDialogue} {@code Object}.
	 */
	public MonkeyDialogue() {
		/**
		 * empty.
		 */
	}

	/**
	 * Constructs a new {@code MonkeyDialogue} {@code Object}.
	 * @param player the player.
	 */
	public MonkeyDialogue(final Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {
		return new MonkeyDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		player("Hey little man, how's it goin'?");
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			npc("Ukkuk oook! Eeek aka, ahh aka gonk.");
			stage++;
			break;
		case 1:
			player(FacialExpression.HALF_GUILTY,"Yeah.");
			stage = 5;
			break;
		case 5:
			end();
			break;
		}
		return true;
	}
	@Override
	public int[] getIds() {
		return new int[] { 2301 };
	}

}

File: Server/src/main/content/region/karamja/dialogue/MonkeyDialogue.kt
package content.region.karamja.dialogue

import core.game.dialogue.DialoguePlugin
import core.game.dialogue.FacialExpression
import core.game.node.entity.npc.NPC
import core.game.node.entity.player.Player
import core.plugin.Initializable
import core.tools.END_DIALOGUE
import core.tools.START_DIALOGUE
import org.rs09.consts.NPCs

@Initializable
class MonkeyDialogue(player: Player? = null) : DialoguePlugin(player) {

    override fun newInstance(player: Player?): DialoguePlugin {
        return MonkeyDialogue(player)
    }

    override fun open(vararg args: Any?): Boolean {
        npc = args[0] as NPC
        playerl(
            FacialExpression.ASKING, "Hey little man, how's it goin'?"
        ).also { stage = START_DIALOGUE }
        return true
    }

    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        when (stage) {
            0 -> npc("Ukkuk oook! Eeek aka, ahh aka gonk.").also { stage++ }

            1 -> playerl(
                FacialExpression.HALF_GUILTY, "Yeah."
            ).also { stage = END_DIALOGUE }
        }
        return true
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.MONKEY_2301)
    }
}